elide-core/src/main/java/com/yahoo/elide/parsers/state/CollectionTerminalState.java

```diff
@@ -71,8 +71,12 @@ public CollectionTerminalState(Class<?> entityClass, Optional<PersistentResource
         DocumentProcessor includedProcessor = new IncludedProcessor();
         includedProcessor.execute(jsonApiDocument, collection, queryParams);
 
-        DocumentProcessor sortProcessor = new SortProcessor();
-        sortProcessor.execute(jsonApiDocument, collection, queryParams);
+        // if we are using pagination, then we are adding sorting directly into the HQL, etc
+        // todo - is this correct?
+        if (requestScope.getPagination().isDefault()) {
```

Comment:
I think the sorting should move out of the Document Processor and into the DataStore.  InMemoryDataStore can preserve some of the functionality of this processor.  Hibernate can then add the sort to the criteria.